frontend/src/components/SegmentModals/SavedSegmentModal/index.tsx

```diff
@@ -29,77 +33,90 @@ export const SavedSegmentModal = ({
 	onSubmit,
 	queryBuilder,
 	shouldUpdate,
-	showModal,
 }: Props) => {
-	const [newSegmentName, setNewSegmentName] = useState(
-		currentSegment?.name ?? '',
-	)
-
-	useEffect(() => {
-		if (shouldUpdate && currentSegment?.name) {
-			setNewSegmentName(currentSegment?.name)
-		} else {
-			setNewSegmentName('')
-		}
-	}, [currentSegment?.name, shouldUpdate])
+	const formStore = useFormStore({
+		defaultValues: {
+			name: currentSegment?.name || '',
+			filters: '',
+		},
+	})
 
 	const handleSubmit = (e: { preventDefault: () => void }) => {
 		e.preventDefault()
 
-		if (!newSegmentName) {
-			return
-		}
-
+		const newSegmentName = formStore.getValue(formStore.names.name)
 		onSubmit(newSegmentName)
 	}
 
 	return (
 		<Modal
-			title={shouldUpdate ? 'Update a Segment' : 'Create a Segment'}
-			visible={showModal}
-			onCancel={onHideModal}
-			style={{ display: 'flex' }}
-			width={500}
-			destroyOnClose
+			title={shouldUpdate ? 'Update Segment' : 'Create Segment'}
+			onClose={onHideModal}
 		>
-			<ModalBody>
-				<form onSubmit={handleSubmit}>
-					<p className={styles.modalSubTitle}>
-						Segments allow you to save search queries that target a
-						specific set of sessions.
-					</p>
-					<div className={styles.queryBuilderContainer}>
-						{queryBuilder}
-					</div>
-					<Input
-						name="name"
-						value={newSegmentName}
-						onChange={(e) => {
-							setNewSegmentName(e.target.value)
-						}}
-						placeholder="Segment Name"
+			<Stack py="8" px="12" style={{ maxWidth: 500 }}>
+				<Form onSubmit={handleSubmit} store={formStore}>
+					<Form.Input
+						name={formStore.names.name}
+						label="Segment Name"
```

Comment:
padding looks a bit tight on the filter tags

<img width="140" alt="Screenshot 2024-01-26 at 1 40 37 PM" src="https://github.com/highlight/highlight/assets/1351531/729dd7d8-e0ef-44af-8056-b4c8c4e5847e">

Reply:
Thanks for flagging! Tried to make it more aligned with figma and event across the modal
![Screenshot 2024-01-29 at 9 55 36 AM](https://github.com/highlight/highlight/assets/17744174/67da3665-d9ed-4c84-985c-9e6a2ac38624)